### DISCUSSION TITLE
permanet public url

### DISCUSSION THREAD

**Discussion by MartinsRepo** [oobabooga/text-generation-webui#3002] [Q&A]
Hi,
can somebody explain what to do to get a permanent public url?
Running on local URL:  http://127.0.0.1:7860
Running on public URL: https://f7105bdd89e126a57e.gradio.live

This share link expires in 72 hours. For free permanent hosting and GPU upgrades (NEW!), check out Spaces: https://huggingface.co/spaces

**Comment by missionfloyd** [comment]
What they mean is by hosting it on huggingface instead of your PC.

You could also use `--listen` instead, and forward port 7860 to your PC. Then you can access it using your public IP address.
